### PR TITLE
Moving GroupBox accessible object to a separate file

### LINKED ISSUE #3229: GroupBoxAccessibleObject class definition should be in a separate file
* .NET Core Version: 

.NET Core 5.0
Version: 5.0.100-alpha.1.20073.10
Commit: 29f4d693a9
Runtime Environment:
OS Name: Windows
OS Version: 10.0.18363
OS Platform: Windows
RID: win10-x64
Base Path: C:\Program Files\dotnet\sdk\5.0.100-alpha1-05536
Host (useful for support):
Version: 5.0.0-alpha.1.20072.3
Commit: c3dc1fdfdc

**Problem description:**

GroupBoxAccessibleObject class definition is located in the same file as a GroupBox control definition.

**Expected behavior:**

GroupBoxAccessibleObject class should be in a separate file and GroupBox class should be partial.

### PR DESCRIPTION
**Pull request by M-Lipin**
Fixes #3229 

## Proposed changes

- Moving GroupBox accessible object to a separate file.
- Minor refactoring and code style changes.
- No functional changes.

## Customer Impact

- Improving development experience for GroupBox control accessibility.

## Regression? 

- No

## Risk

- Minimal

### Before

image

### After

image

## Test methodology 

- Manual testing.
- Separated accessible object should be covered by unit test.
- UI automation.

## Accessibility testing  

- Inspect too.
- Accessibility client apps.
 

## Test environment(s) 

.NET Core 5.0
Version: 5.0.100-alpha.1.20073.10
Commit: 29f4d693a9
Runtime Environment:
OS Name: Windows
OS Version: 10.0.18363
OS Platform: Windows
RID: win10-x64
Base Path: C:\Program Files\dotnet\sdk\5.0.100-alpha1-05536
Host (useful for support):
Version: 5.0.0-alpha.1.20072.3
Commit: c3dc1fdfdc

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3230)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/GroupBox.GroupBoxAccessibleObject.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#nullable disable
+
+using System.Runtime.InteropServices;
+using static Interop;
+
+namespace System.Windows.Forms
+{
+    public partial class GroupBox
+    {
+        [ComVisible(true)]
```
Comment (weltkante) [inline src/System.Windows.Forms/src/System/Windows/Forms/GroupBox.GroupBoxAccessibleObject.cs:14]: don't add ComVisible, I don't think its needed

File: src/System.Windows.Forms/src/System/Windows/Forms/GroupBox.GroupBoxAccessibleObject.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/GroupBox.GroupBoxAccessibleObject.cs:5]: We want NRT

File: src/System.Windows.Forms/src/System/Windows/Forms/GroupBox.GroupBoxAccessibleObject.cs (hunk 1 above)
Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/GroupBox.GroupBoxAccessibleObject.cs:14]: Let's tackle all `ComVisible` together once we reach a consensus on it, tracking in https://github.com/dotnet/winforms/issues/1878.

Comment (weltkante) [reply src/System.Windows.Forms/src/System/Windows/Forms/GroupBox.GroupBoxAccessibleObject.cs:14]: Yeah sorry this comment was from before I realized it doesn't generate the same GUID as Desktop Framework. Personally I'd prefer not to add new ComVisible attributes though, but it doesn't cause immediate problems if you do (it just doesn't have anny purpose either as far as I'm aware).

Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/GroupBox.GroupBoxAccessibleObject.cs:12]: Thanks!

File: src/System.Windows.Forms/src/System/Windows/Forms/GroupBox.GroupBoxAccessibleObject.cs [hunk 3]
```diff
@@ -2,8 +2,6 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
```
Comment (M-Lipin) [inline src/System.Windows.Forms/src/System/Windows/Forms/GroupBox.GroupBoxAccessibleObject.cs]: Thank you!

File: src/System.Windows.Forms/src/System/Windows/Forms/GroupBox.GroupBoxAccessibleObject.cs (hunk 2 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/GroupBox.GroupBoxAccessibleObject.cs]: Thanks for fixing it!

**Pull request merged**